source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -73,20 +71,35 @@ FilterConfig::FilterConfig(
     throw EnvoyException("local rate limit token bucket must be set for per filter configs");
   }
 
+  absl::Status creation_status;
+  rate_limit_config_ = std::make_unique<Filters::Common::RateLimit::RateLimitConfig>(
+      config.rate_limits(), context, creation_status);
+  THROW_IF_NOT_OK_REF(creation_status);
+
+  if (rate_limit_config_->empty()) {
+    if (!config.descriptors().empty()) {
+      ENVOY_LOG_FIRST_N(
+          warn, 20,
+          "'descriptors' are set for local rate limit filter but no 'rate_limits' "
+          "are configured in the filter config. Please use the 'rate_limits' field "
+          "in filter config to instead of the 'rate_limits' field in the route config.");
+    }
+  }
```

Comment:
Shouldn't this cause a config load failure?

Reply:
Nope. This warning will notice the users to switch the `rate_limits` from route to the embedded `rate_limits` of filter config.

But before we removed the support to the route `rate_limits` completely in the filter, we shouldn't reject the configuration that without embedded `rate_limits`.

Reply:
Then sorry I don't understand this warning. Can you make it more clear?

Reply:
Let me have a try make it more clear.

Reply:
I have updated the log message. Hope it makes sense for you. The core points are:

1. `descriptors` only makes sense when `rate_limits` (of filter config or route) is configured.
2. Now, the `rate_limits` in filter config is not set, may be it's set in the route. Please take the `rate_limits` of filter config as priority because we may remove the support of `rate_limits` of route config from the local rate limit filter in the future.